azurerm/resource_arm_logic_app_workflow.go

```diff
@@ -0,0 +1,469 @@
+						"recurrence": {
+							Type:     schema.TypeList,
+							Required: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"name": {
+										Type:     schema.TypeString,
+										Required: true,
+									},
+									"frequency": {
+										Type:     schema.TypeString,
+										Required: true,
+										ValidateFunc: validation.StringInSlice([]string{
+											"Month",
+											"Week",
+											"Day",
+											"Hour",
+											"Minute",
+											"Hour",
+											"Second",
+										}, true),
+									},
+									"interval": {
+										Type:     schema.TypeInt,
+										Required: true,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+
+			"action": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Computed: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"http": {
+							Type:     schema.TypeList,
+							Optional: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"name": {
+										Type:     schema.TypeString,
+										Required: true,
+									},
+									"method": {
+										Type:     schema.TypeString,
+										Required: true,
+									},
+									"uri": {
+										Type:     schema.TypeString,
+										Required: true,
+									},
+								},
+							},
+						},
+
+						"function": {
+							Type:     schema.TypeList,
+							Optional: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"name": {
+										Type:     schema.TypeString,
+										Required: true,
+									},
+									"function_id": {
+										Type:     schema.TypeString,
+										Required: true,
+									},
+									"body": {
+										Type:     schema.TypeString,
+										Required: true,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+
+			"tags": tagsSchema(),
+
+			"access_endpoint": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceArmLogicAppWorkflowCreateUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).logicWorkflowsClient
+	ctx := meta.(*ArmClient).StopContext
+
+	log.Printf("[INFO] preparing arguments for Logic App Workflow creation.")
+
+	name := d.Get("name").(string)
+	resourceGroup := d.Get("resource_group_name").(string)
+	location := azureRMNormalizeLocation(d.Get("location").(string))
+	definition := expandLogicAppWorkflowDefinition(d)
+	parameters := expandLogicAppWorkflowParameters(d)
+	tags := d.Get("tags").(map[string]interface{})
+
+	properties := logic.Workflow{
+		Location: utils.String(location),
+		WorkflowProperties: &logic.WorkflowProperties{
+			Definition: definition,
+			Parameters: parameters,
+		},
+		Tags: expandTags(tags),
+	}
+
+	_, err := client.CreateOrUpdate(ctx, resourceGroup, name, properties)
+	if err != nil {
+		return err
```

Comment:
Error message is not consistent with other `return` statements. Provide more detailed info.